xarray/coding/times.py

```diff
@@ -579,20 +579,23 @@ def _numbers_to_timedelta(
 ) -> np.ndarray:
     """Transform numbers to np.timedelta64."""
     # keep NaT/nan mask
-    nan = np.isnan(flat_num) | (flat_num == np.iinfo(np.int64).min)
+    if flat_num.dtype.kind == "f":
```

Comment:
That's a bit more explicit. The asarray-wrapping is needed for upstream-dev to keep 0d-arrays an array.